community/kernel/src/main/java/org/neo4j/kernel/impl/store/StandardDynamicRecordAllocator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store;
+
+import org.neo4j.kernel.impl.store.id.IdSequence;
+import org.neo4j.kernel.impl.store.record.DynamicRecord;
+
+public class StandardDynamicRecordAllocator implements DynamicRecordAllocator
+{
+    protected final IdSequence idGenerator;
+    private final int dataSize;
+
+    public StandardDynamicRecordAllocator( IdSequence idGenerator, int dataSize )
+    {
+        this.idGenerator = idGenerator;
+        this.dataSize = dataSize;
+    }
+
+    @Override
+    public int getRecordDataSize()
+    {
+        return dataSize;
+    }
+
+    @Override
+    public DynamicRecord nextRecord()
+    {
+        return allocateRecord( idGenerator.nextId() );
+    }
+
+    public static DynamicRecord allocateRecord( long id )
```

Comment:
this is looking really strange: we own id generator and only we should be responsible for next id generation in the same time there is a public method that can accept id from somebody else.
And single usage says that that method can be avoided

Reply:
It's a static method allocating a `DynamicRecord` with a certain id, setting its `created` and `inUse` flags. What's strage with that? :)

Reply:
The Strange thing is not what it does, by why it publicly exposed in a current way.
StandardDynamicRecordAllocator has a reference to `idGenerator` and responsible for generating of correct `nextRecord` with an appropriate id that should be allocated by `idGenerator`. First strange thing - who, if not id generator, can generate a correct id for you. Another one is it's single usage in AbstractDynamicStore: `StandardDynamicRecordAllocator.allocateRecord( nextId() )` where that next id will be generated from idGenerator as well, so why to expose it at all

Reply:
It's not a single usage, there's also `StandardDynamicRecordAllocator` (instance) and its subclass `RelativeIdRecordAllocator`, isn't there?